stat/file.go

```diff
@@ -56,27 +56,29 @@ const (
 // FileStats contains all the file stats needed by wrstat, interpreted in our
 // custom way.
 type FileStats struct {
-	QuotedPath string
-	Size       int64
-	UID        uint32
-	GID        uint32
-	Atim       int64
-	Mtim       int64
-	Ctim       int64
-	Type       FileType
-	Ino        uint64
-	Nlink      uint64
-	Dev        uint64
+	Path  string
+	Size  int64
+	UID   uint32
+	GID   uint32
+	Atim  int64
+	Mtim  int64
+	Ctim  int64
+	Type  FileType
+	Ino   uint64
+	Nlink uint64
+	Dev   uint64
 }
 
-// ToString produces our special format for describing the stats of a file. It
-// is \n terminated and ready to be written to a file.
-func (fs *FileStats) ToString() string {
-	return fmt.Sprintf(
-		"%s\t%d\t%d\t%d\t%d\t%d\t%d\t%s\t%d\t%d\t%d\n",
-		fs.QuotedPath, fs.Size, fs.UID, fs.GID,
+// WriteTo produces our special format for describing the stats of a file. It
+// is \n terminated and writes to the given Writer.
+func (fs *FileStats) WriteTo(w io.Writer) (int64, error) {
+	n, err := fmt.Fprintf(w,
+		"%q\t%d\t%d\t%d\t%d\t%d\t%d\t%s\t%d\t%d\t%d\n",
+		fs.Path, fs.Size, fs.UID, fs.GID,
```

Comment:
Why was the property renamed from QuotedPath to Path, but not quoted when we finally output?

Reply:
It is quoted, we're using the format string to do the quoting.